trident/absorption_spectrum/absorption_spectrum.py

```diff
@@ -612,14 +733,39 @@ def _add_lines_to_spectrum(self, field_data, use_peculiar_velocity,
                 # value at the ends to assure that the wings of a line have been
                 # fully resolved.
                 while True:
-                    left_index = (center_index[i] - window_width_in_bins//2)
-                    right_index = (center_index[i] + window_width_in_bins//2)
-                    n_vbins = (right_index - left_index) * n_vbins_per_bin[i]
+
+                    # calculate wavelength window
+                    if self._auto_lambda and self.lambda_field is None:
+                        my_lambda_min = lambda_obs[i] - \
+                          window_width_in_bins * self.bin_width / 2
+                        # round off to multiple of bin_width
+                        my_lambda_min = self.bin_width * \
+                          np.ceil(my_lambda_min / self.bin_width)
+                        my_lambda = my_lambda_min + \
+                          self.bin_width * np.arange(window_width_in_bins)
+
+                    else:
+                        my_lambda = self.lambda_field
+
+                    # we want to know the bin index in the lambda_field array
+                    # where each line has its central wavelength after being
+                    # redshifted.  however, because we don't know a priori how wide
+                    # a line will be (ie DLAs), we have to include bin indices
+                    # *outside* the spectral range of the AbsorptionSpectrum
+                    # object.  Thus, we find the "equivalent" bin index, which
+                    # may be <0 or >the size of the array.  In the end, we deposit
+                    # the bins that actually overlap with the AbsorptionSpectrum's
+                    # range in lambda.
+
+                    left_index, center_index, right_index = \
+                      get_bin_indices(my_lambda, self.bin_width,
+                                      lambda_obs[i], window_width_in_bins)
+                    n_vbins = window_width_in_bins * n_vbins_per_bin[i]
 
                     # the array of virtual bins in lambda space
                     vbins = \
-                        np.linspace(self.lambda_min + self.bin_width.d * left_index,
-                                    self.lambda_min + self.bin_width.d * right_index,
+                        np.linspace(my_lambda.d[0] + self.bin_width.d * left_index,
```

Comment:
Will `my_lambda` always have the `.d` attribute? I wasn't able to fully trace the code to confirm this. @brittonsmith?

Reply:
Good question. The answer should be yes. `my_lambda` will only be made in two places: 
1. by the function `_create_lambda_field` which explicitly creates a `YTArray`
2. on roughly line 744 of this file, where it uses `self.bin_width`, which will always have units as per its definition in the `__init__` function.

Reply:
perfect. Thanks for the clarification!